src/update_engine/payload_processor.cc

```diff
@@ -52,15 +52,39 @@ void LogPartitionInfo(const DeltaArchiveManifest& manifest) {
 
 }  // namespace {}
 
+PayloadProcessor::PayloadProcessor(PrefsInterface* prefs, InstallPlan* install_plan)
+  : partition_performer_(prefs, install_plan->install_path),
+    prefs_(prefs),
+    install_plan_(install_plan),
+    manifest_valid_(false),
+    manifest_metadata_size_(0),
+    next_operation_num_(0),
+    buffer_offset_(0),
+    last_updated_buffer_offset_(std::numeric_limits<uint64_t>::max()),
+    public_key_path_(kUpdatePayloadPublicKeyPath),
+    num_total_operations_(0) {
+}
+
+int PayloadProcessor::Open() {
+  return partition_performer_.Open();
+}
+
 int PayloadProcessor::Close() {
-  int err = delta_performer_.Close();
+  int err = partition_performer_.Close();
   LOG_IF(ERROR, !hash_calculator_.Finalize()) << "Unable to finalize the hash.";
   if (!buffer_.empty()) {
     LOG(ERROR) << "Called Close() while buffer not empty!";
     if (err == 0) {
       err = -EBUSY;
     }
   }
+  if (next_operation_num_ != num_total_operations_) {
```

Comment:
After `delta_performer_.Close(); hash_calculator_.Finalize()` everything here is just checking for bugs so I don't think it really matters what order any of it happens.